File: ApexFit/Models/JournalEntry.swift
import Foundation
import SwiftData

@Model
final class JournalEntry {
    var id: UUID
    var date: Date
    var completedAt: Date?
    var isComplete: Bool
    var streakDays: Int
    var createdAt: Date

    var userProfile: UserProfile?

    @Relationship(deleteRule: .cascade, inverse: \JournalResponse.journalEntry)
    var responses: [JournalResponse]

    init(date: Date) {
        self.id = UUID()
        self.date = Calendar.current.startOfDay(for: date)
        self.isComplete = false
        self.streakDays = 0
        self.createdAt = Date()
        self.responses = []
    }

    var responseCount: Int {
        responses.count
    }

    var formattedDate: String {
        date.formatted(date: .abbreviated, time: .omitted)
    }
}

File: android/core/model/src/main/kotlin/com/apexfit/core/model/JournalEntry.kt
package com.apexfit.core.model

data class JournalEntry(
    val id: String = java.util.UUID.randomUUID().toString(),
    val date: Long, // epoch millis, start of day
    val completedAt: Long? = null,
    val isComplete: Boolean = false,
    val streakDays: Int = 0,
    val createdAt: Long = System.currentTimeMillis(),
    val userProfileId: String? = null,
    val responses: List<JournalResponse> = emptyList(),
) {
    val responseCount: Int get() = responses.size
}
